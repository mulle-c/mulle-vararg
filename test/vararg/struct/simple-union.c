#include <mulle-vararg/mulle-vararg.h>
#include <stdio.h>


union some_union
{
   int   x;
   char  y[ 5];  // not so nice alignment
};


struct
{
   long long          a;
   union some_union   b;
   union some_union   c;
} value =
{
   -1848,
   {
      .y = { '1', '8', '4', '8', 0}
   },
   {
      .y = { '1', '8', '4', '9', 0}
   }
};


int   main( void)
{
   mulle_vararg_list   list;
   union some_union    v;

   mulle_vararg_start( list, value.a);

   v = mulle_vararg_next_union( list, union some_union);
   printf( "%s\n", v.y);

   v = mulle_vararg_next_union( list, union some_union);
   printf( "%s\n", v.y);

   mulle_vararg_end( list);

   return( 0);
}
