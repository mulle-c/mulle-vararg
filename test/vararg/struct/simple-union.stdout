1848
1849
